libparsec/src/client.rs

```diff
@@ -120,7 +122,7 @@ pub async fn client_start(
 
     // 3) Actually start the client
 
-    let on_event = OnEventCallbackPlugged::new(on_event_callback);
+    let on_event = OnEventCallbackPlugged::new(handle, on_event_callback);
```

Comment:
```suggestion
    let on_event = OnEventCallbackPlugged::new(initializing.handle(), on_event_callback);
```

`initializing.handle` filed is private (and owner shouldn't be able to modify it), so you should add a `InitializingGuard::handle(&self) ->`Handle ` getter

Reply:
done